src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -111,63 +113,193 @@ private function generate_stock_status_count_query( $status, $product_query_sql,
+			$counts = $this->get_terms_list( $filtered_attribute );
+
+			return array_map( 'absint', wp_list_pluck( $counts, 'term_count', 'term_count_id' ) );
+		}
+
+		$where_clause = '';
+		if ( ! empty( $min_price ) || ! empty( $max_price ) || ! empty( $rating ) || ! empty( $stock_status ) ) {
+			$product_metas = [
+				'min_price'      => $min_price,
+				'max_price'      => $max_price,
+				'average_rating' => $rating,
+				'stock_status'   => $stock_status,
+			];
+
+			$filtered_products_by_metas           = $this->get_product_by_metas( $product_metas );
+			$formatted_filtered_products_by_metas = implode( ',', array_map( 'intval', $filtered_products_by_metas ) );
+
+			if ( ! empty( $formatted_filtered_products_by_metas ) ) {
+				if ( ! empty( $rating ) ) {
+					$where_clause .= sprintf( ' AND product_attribute_lookup.product_or_parent_id IN (%1s)', $formatted_filtered_products_by_metas );
+				} else {
+					$where_clause .= sprintf( ' AND product_attribute_lookup.product_id IN (%1s)', $formatted_filtered_products_by_metas );
+				}
+			} else {
+				$counts = $this->get_empty_terms_list( $filtered_attribute );
+
+				return array_map( 'absint', wp_list_pluck( $counts, 'term_count', 'term_count_id' ) );
+			}
+		}
+
+		$join_type = 'LEFT';
+		foreach ( $attributes_data as $attribute ) {
+			$filtered_terms = $attribute['slug'] ?? '';
+
+			if ( empty( $filtered_terms ) ) {
+				continue;
+			}
 
-		return array_map( 'absint', wp_list_pluck( $results, 'term_count', 'term_count_id' ) );
+			$taxonomy = $attribute['attribute'] ?? '';
+			$term_ids = [];
+			if ( in_array( $taxonomy, wc_get_attribute_taxonomy_names(), true ) ) {
+				foreach ( $filtered_terms as $filtered_term ) {
+					$term = get_term_by( 'slug', $filtered_term, $taxonomy );
+					if ( is_object( $term ) ) {
+						$term_ids[] = $term->term_id;
+					}
+				}
+			}
+
+			if ( empty( $term_ids ) ) {
+				continue;
+			}
+
+			foreach ( $calculate_attribute_counts as $calculate_attribute_count ) {
+				if ( ! isset( $calculate_attribute_count['taxonomy'] ) && ! isset( $calculate_attribute_count['query_type'] ) ) {
+					continue;
+				}
+
+				$query_type                           = $calculate_attribute_count['query_type'];
+				$filtered_products_by_terms           = $this->get_product_by_filtered_terms( $calculate_attribute_count['taxonomy'], $term_ids, $query_type );
+				$formatted_filtered_products_by_terms = implode( ',', array_map( 'intval', $filtered_products_by_terms ) );
+
+				if ( ! empty( $formatted_filtered_products_by_terms ) ) {
+					$where_clause .= sprintf( ' AND product_attribute_lookup.product_or_parent_id IN (%1s)', $formatted_filtered_products_by_terms );
+				}
+
+				if ( $calculate_attribute_count['taxonomy'] === $filtered_attribute ) {
+					$join_type = 'or' === $query_type ? 'LEFT' : 'INNER';
+				}
+			}
+		}
+
+		global $wpdb;
+		$counts = $wpdb->get_results(
```

Comment:
Can we format this block of query code with indentation because VSCode is showing formatting alerts.

Reply:
Could you give an example, please? My VSCode is not showing any formatting alerts, and things are a-ok regarding coding standards (PHPCS is not throwing any errors); maybe that's coming from an extension?

Reply:
Yeah I think it has mixed of tabs and spaces. ![kfxLup.png](https://user-images.githubusercontent.com/2132595/230418787-63f82edd-e861-4c87-9be1-8c7af0c90a7a.png)

Reply:
Gotcha: I'm a bit in the dark here as none of my IDEs are flagging this problem, but pushed this formatting update just in case  https://github.com/woocommerce/woocommerce-blocks/pull/8599/commits/cb40f042651b931875c652dc5136c6e4797a3bd8

Reply:
I pushed up an update to fix that. Thanks!